networkit/cpp/centrality/PageRank.cpp

```diff
@@ -71,17 +72,39 @@ void PageRank::run() {
 
     handler.assureRunning();
 
-    const auto sum = G.parallelSumForNodes([&](const node u) { return scoreData[u]; });
+    if (!normalized) {
+        const auto sum = G.parallelSumForNodes([&](const node u) { return scoreData[u]; });
+
+        // make sure scoreData sums up to 1
+        assert(!Aux::NumericTools::equal(sum, 0.0, 1e-15));
+        G.parallelForNodes([&](const node u) { scoreData[u] /= sum; });
+    } else {
+        // calculate sum of the Pagerank of dangling Nodes for normalization
+        const auto sum = G.parallelSumForNodes([&](const node u) {
+            if (G.degree(u) == 0.0)
+                return scoreData[u];
+            return 0.0;
+        });
+
+        const auto normFactor = (1.0 / n) * (tol + ((1.0 - tol) * sum));
```

Comment:
I think the usage of `tol` is wrong here. According to the paper, `r_low`is computed with the teleport probabilty (or `1-damp`). This also could explain the high values for normalized page rank.

Reply:
You're right. Eps there is not the tolerance but 1-damp.

Reply:
Better, but still not correct. Admitted - terms are bit confusing. If you look at the rest of the implementation of PageRank and compare this with the paper, you see that `damp` (from our code) = `1 - eps` (even though it is called damping factor in the paper). You can see this for example when looking at the computation of the final teleport prob.